dsl.go

```diff
@@ -144,8 +148,9 @@ func (genericMock *GenericMock) GetInvocationParams(methodInvocations []MethodIn
 
 func (genericMock *GenericMock) methodInvocations(methodName string, params []Param, matchers []Matcher) []MethodInvocation {
 	var invocations []MethodInvocation
-	if _, exists := genericMock.mockedMethods[methodName]; exists {
-		for _, invocation := range genericMock.mockedMethods[methodName].invocations {
+	if method, exists := genericMock.mockedMethods[methodName]; exists {
+		method.Lock()
```

Comment:
Lock not needed if we don't do invocations and verifications concurrently.

Reply:
I think you're right, but I encountered a race condition in this place after upgrade to https://github.com/petergtz/pegomock/commit/8c753638498b1f3f5fa7eda74630a4b24bfac08d

I, most likely, incorrectly made the logic of the function being tested, since the function that I call and want to check is run in goroutine.